pkg/cmd/server/api/v1/types.go

```diff
@@ -732,6 +732,9 @@ type RequestHeaderIdentityProvider struct {
 
 	// ClientCA is a file with the trusted signer certs.  If empty, no request verification is done, and any direct request to the OAuth server can impersonate any identity from this provider, merely by setting a request header.
 	ClientCA string `json:"clientCA"`
+	// ClientCommonNames is an optional list of common names to require a match from. If empty, any client certificate validated against the clientCA bundle is considered authoritative.
+	ClientCommonNames []string `json:"clientCommonNames"`
```

Comment:
> I might make it clientAuthorizedCommonNames to indicate that it's a list of the ones that will be accepted.

I think that's unnecessary.  We don't specify the allowed side of an option unless we have a corresponding deny.   Take the preceding option as an example.  `authorizedClientCA` instead of `clientCA` is extraneous, because you don't make a blacklist of CAs (usually).